snmp/__init__.py

```diff
@@ -0,0 +1,315 @@
+        """
+
+        from bin.smarthome import VERSION
+        if '.'.join(VERSION.split('.', 2)[:2]) <= '1.5':
+            self.logger = logging.getLogger(__name__)
+
+        self.logger.debug("init {}".format(__name__))
+
+        # get the parameters for the plugin (as defined in metadata plugin.yaml):
+        self.instance = self.get_parameter_value('instance')         # the instance of the plugin for questioning multiple smartmeter
+        self.cycle = self.get_parameter_value('cycle')               # the frequency in seconds how often the query shoud be done
+        self.host = self.get_parameter_value('snmp_host')            # IP Adress of the network device to be queried
+        self.port = self.get_parameter_value('snmp_port')            # Port for SNMP queries
+        self.community = self.get_parameter_value('snmp_community')  # SNMP Community
+
+        # Initialization code goes here
+        self._items = {}
+        """
+        self._items will contain something like
+        {'1.3.6.1.4.1.24681.1.2.5.0': {'temp': {'item': Item: netzwerk.qnap_ts251.cpu_temp}}, '1.3.6.1.4.1.24681.1.2.1.0': {'precent': {'item': Item: netzwerk.qnap_ts251.cpu_usage}}, '1.3.6.1.4.1.24681.1.2.6.0': {'temp': {'item': Item: netzwerk.qnap_ts251.system_temp}}}
+        """
+
+        # log
+        self.logger.debug("Instance {} of SNMP configured to use host '{}' with update cycle of {} seconds and Community {}".format(self.instance if self.instance else 0, self.host, self.cycle, self.community))
+
+        # give some info to the user via webinterface
+        self.init_webinterface()
+
+        # init_complete to True
+        self._init_complete = True
+
+    def run(self):
+        """
+        Run method for the plugin
+        """
+        self.logger.debug("Run method called")
+        self.alive = True
+        self.scheduler_add('update', self._poll_cycle, cycle=self.cycle)
+
+    def stop(self):
+        """
+        Stop method for the plugin
+        """
+        self.alive = False
+        self.logger.debug("Stop method called")
+        self.scheduler_remove('update')
+
+    def parse_item(self, item):
+        """
+        Default plugin parse_item method. Is called when the plugin is initialized.
+        The plugin can, corresponding to its attribute propwords, decide what to do with
+        the item in future, like adding it to an internal array for future reference
+        :param item:    The item to process.
+        """
+        if not self.has_iattr(item.conf, 'snmp_oid'):
+            return
+        oid = self.get_iattr_value(item.conf, 'snmp_oid')
+
+        if not self.has_iattr(item.conf, 'snmp_prop'):
+            self.logger.warning("SNMP: No snmp_prop for {0} defined, set to standard".format(item.id()))
+            prop = 'std'
+        else:
+            prop = self.get_iattr_value(item.conf, 'snmp_prop').lower()
+
+        table = self._items
```

Comment:
what is the purpose of using table here as reference and not directly self._items instead?

Reply:
changed; now just self._items will be used